scripts/updatedb.js

```diff
@@ -0,0 +1,638 @@
+// fetches and converts maxmind lite databases
+
+'use strict'
+
+const utils = require('geoip-lite/lib/utils')
+
+const userAgent = 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.36 Safari/537.36'
+
+const fs = require('fs')
+const http = require('http')
+const https = require('https')
+const path = require('path')
+const zlib = require('zlib')
+
+fs.existsSync = fs.existsSync || path.existsSync
+
+const async = require('async')
+const colors = require('colors') // eslint-disable-line no-unused-vars
+const iconv = require('iconv-lite')
+const lazy = require('lazy')
+const rimraf = require('rimraf').sync
+const yauzl = require('yauzl')
+const Address6 = require('ip-address').Address6
+const Address4 = require('ip-address').Address4
+
+const args = process.argv.slice(2)
+let licenseKey = args.find(function (arg) {
+  return arg.match(/^license_key=[a-zA-Z0-9]+/) !== null
+})
+if (typeof licenseKey === 'undefined' && typeof process.env.LICENSE_KEY !== 'undefined') {
+  licenseKey = 'license_key=' + process.env.LICENSE_KEY
+}
+const dataPath = path.join(__dirname, '..', 'data')
+const tmpPath = path.join(__dirname, '..', 'tmp')
+const countryLookup = {}
+const cityLookup = {}
+const databases = [
+  {
+    type: 'country',
+    url: 'https://geoip.maxmind.com/app/geoip_download?edition_id=GeoIP2-Country-CSV&suffix=zip&' + licenseKey,
+    checksum: 'https://geoip.maxmind.com/app/geoip_download?edition_id=GeoIP2-Country-CSV&suffix=zip.md5&' + licenseKey,
+    fileName: 'GeoIP2-Country-CSV.zip',
+    src: [
+      'GeoIP2-Country-Locations-en.csv',
+      'GeoIP2-Country-Blocks-IPv4.csv',
+      'GeoIP2-Country-Blocks-IPv6.csv'
+    ],
+    dest: [
+      '',
+      'geoip-country.dat',
+      'geoip-country6.dat'
+    ]
+  },
+  {
+    type: 'city',
+    url: 'https://geoip.maxmind.com/app/geoip_download?edition_id=GeoIP2-City-CSV&suffix=zip&' + licenseKey,
+    checksum: 'https://geoip.maxmind.com/app/geoip_download?edition_id=GeoIP2-City-CSV&suffix=zip.md5&' + licenseKey,
+    fileName: 'GeoIP2-City-CSV.zip',
+    src: [
+      'GeoIP2-City-Locations-en.csv',
+      'GeoIP2-City-Blocks-IPv4.csv',
+      'GeoIP2-City-Blocks-IPv6.csv'
+    ],
+    dest: [
+      'geoip-city-names.dat',
+      'geoip-city.dat',
+      'geoip-city6.dat'
+    ]
+  }
+]
+
+function mkdir (name) {
+  const dir = path.dirname(name)
+  if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir)
+  }
+}
+
+// Ref: http://stackoverflow.com/questions/8493195/how-can-i-parse-a-csv-string-with-javascript
+// Return array of string values, or NULL if CSV string not well formed.
+// Return array of string values, or NULL if CSV string not well formed.
+
+function tryFixingLine (line) {
```

Comment:
maybe we can use https://www.npmjs.com/package/csv-parse as it's well tested and is actively maintained? sorry for asking to remove this part, I know it took quite some time to figure it out :(